plugins/bf4/bf4.cpp

```diff
@@ -0,0 +1,229 @@
+    }
+
+    bool ok, state, squad_leader;
+    char serverid[37], host[22], team[3];
+    BYTE squad;
+
+    // Server ID pointers
+    BYTE *serverid_base = peekProc<BYTE *>(pModule + 0x02210658);
+    if (!serverid_base) return false;
+    BYTE *serverid_offset_0 = peekProc<BYTE *>(serverid_base + 0x18);
+    if (!serverid_offset_0) return false;
+    BYTE *serverid_offset_1 = peekProc<BYTE *>(serverid_offset_0 + 0x28);
+    if (!serverid_offset_1) return false;
+    BYTE *serverid_offset = peekProc<BYTE *>(serverid_offset_1 + 0x350);
+    if (!serverid_offset) return false;
+
+    // Squad pointers
+    BYTE *squad_base = peekProc<BYTE *>(pModule + 0x02210D68);
+    if (!squad_base) return false;
+    BYTE *squad_offset_0 = peekProc<BYTE *>(squad_base + 0x570);
+    if (!squad_offset_0) return false;
+    BYTE *squad_offset_1 = peekProc<BYTE *>(squad_offset_0 + 0x100);
+    if (!squad_offset_1) return false;
+    BYTE *squad_offset_2 = peekProc<BYTE *>(squad_offset_1 + 0x58);
+    if (!squad_offset_2) return false;
+
+    // Peekproc and assign game addresses to our containers, so we can retrieve positional data
+    ok = peekProc((BYTE *) pModule + 0x21CAFF0, &state, 1) && // Magical state value: 0 when in-game and 1 when in menu/dead.
+            peekProc((BYTE *) pModule + 0x21C6D40, avatar_pos, 12) && // Avatar Position values (X, Y and Z).
+            peekProc((BYTE *) pModule + 0x21CAF80, camera_pos, 12) && // Camera Position values (X, Y and Z).
+            peekProc((BYTE *) pModule + 0x21CAF60, avatar_top, 12) && // Avatar Top Vector values (X, Y and Z).
+            peekProc((BYTE *) pModule + 0x21CAF70, avatar_front, 12) && // Avatar Front Vector values (X, Y and Z).
+            peekProc((BYTE *) serverid_offset, serverid) && // Server ID (36 characters).
+            peekProc((BYTE *) pModule + 0x21B80C0, host) && // Host value: "IP:Port" when in a server, "bot" when loading and empty when it's hidden.
+            peekProc((BYTE *) pModule + 0x24AFAE5, team) && // Team value: US (United States); RU (Russia); CH (China).
+            peekProc((BYTE *) squad_offset_2 + 0x230, squad) && // Squad value: 0 (not in a squad); 1 (Alpha); 2 (Bravo); 3 (Charlie)... 26 (Zulu).
+            peekProc((BYTE *) squad_offset_2 + 0x234, squad_leader); // Team value: US (United States); RU (Russia); CH (China).
+
+    // This prevents the plugin from linking to the game in case something goes wrong during values retrieval from memory addresses
+    if (! ok) {
+        return false;
+    }
+
+    if (state) { // If not in-game
+        context.clear(); // Clear context
+        identity.clear(); // Clear identity
+        // Set vectors values to 0.
+        for (int i=0;i<3;i++) {
+            avatar_pos[i] = avatar_front[i] = avatar_top[i] = camera_pos[i] =  camera_front[i] = camera_top[i] = 0.0f;
+        }
+
+        return true; // This tells Mumble to ignore all vectors.
+    }
+    
```

Comment:
In the code below, you are using the C strings `ServerID`, `host` and `team` as-is.

However, if they (for some reason) do not contain a NUL byte, you'll crash Mumble.

Instead, you should probably convert them to std::string before using them.

    std::string sserverid(serverid, sizeof(serverid));
    std::string shost(host, sizeof(host));
    std::string steam(team, sizeof(team));

and use those names in the code below.